src/btree/bt_delete.c

```diff
@@ -204,10 +204,26 @@ __wt_delete_page_rollback(WT_SESSION_IMPL *session, WT_REF *ref)
 			 */
 			__wt_free(session, ref->page_del->update_list);
 			__wt_free(session, ref->page_del);
-			goto done;
+			return;
 		}
+		/*
+		 * We wait for the change in page state, yield before retrying,
+		 * and if we've yielded enough times, start sleeping so we don't
+		 * burn CPU to no purpose.
+		 */
+		if (yield_count < WT_THOUSAND) {
+			yield_count++;
+			__wt_yield();
+			continue;
+		}
+
+		sleep_count = WT_MIN(sleep_count + WT_THOUSAND,
+		    10 * WT_THOUSAND);
+		WT_STAT_CONN_INCRV(session, page_del_rollback_blocked,
+		    sleep_count);
+		__wt_sleep(0, sleep_count);
```

Comment:
I have used a new inline function as per your suggestion.
1. function name is different as s_string is flashing for word "backoff"
2. move the stat increment outside the inline function as the stat variable is identified by name.
3. minimally used this function in bt_read : __wt_page_in_func, as otherwise it is not looking pretty.